src/components/CompareResults/RetriggerButton.tsx

```diff
@@ -124,26 +129,80 @@ function SignInModal(props: SignInModalProps) {
   );
 }
 
+function RetriggerCountSelect({
+  prefix,
+  label,
+}: {
+  prefix: string;
+  label: string;
+}) {
+  return (
+    <FormControl sx={{ width: '100%' }}>
+      <InputLabel id={`${prefix}-retrigger-count-label`}>{label}</InputLabel>
+      <Select
+        labelId={`${prefix}-retrigger-count-label`}
+        name={`${prefix}-retrigger-count`}
+        defaultValue={0}
+        label={label}
+        sx={{ height: 32 }}
+      >
+        <MenuItem value={0}>0</MenuItem>
+        <MenuItem value={1}>1</MenuItem>
+        <MenuItem value={2}>2</MenuItem>
+        <MenuItem value={3}>3</MenuItem>
+        <MenuItem value={4}>4</MenuItem>
+        <MenuItem value={5}>5</MenuItem>
+      </Select>
+    </FormControl>
+  );
+}
+
 type RetriggerModalProps = {
   open: boolean;
   onClose: () => unknown;
   onRetriggerClick: (times: { baseTimes: number; newTimes: number }) => unknown;
 };
 
-function RetriggerModal({ open, onClose }: RetriggerModalProps) {
+function RetriggerModal(props: RetriggerModalProps) {
+  const onFormSubmit = (e: React.FormEvent<HTMLFormElement>) => {
+    e.preventDefault();
+    const formData = new FormData(e.currentTarget);
+
+    const baseCount = formData.get('base-retrigger-count');
+    const newCount = formData.get('new-retrigger-count');
+    props.onRetriggerClick({
+      baseTimes: baseCount ? +baseCount : 0,
+      newTimes: newCount ? +newCount : 0,
+    });
+  };
+
   return (
-    <Modal
-      open={open}
-      onClose={onClose}
-      aria-labelledby='retrigger-modal-title'
-      sx={{ display: 'flex' }}
+    <CenteredModal
+      open={props.open}
+      onClose={props.onClose}
+      ariaLabelledby='retrigger-modal-title'
+      paperStyle={{ display: 'flex', flexDirection: 'column', gap: 2 }}
     >
-      <Box>
-        <Typography id='retrigger-modal-title' component='h2'>
-          Configure the number of retriggers
-        </Typography>
-      </Box>
-    </Modal>
+      <Typography id='retrigger-modal-title' component='h2' variant='h2'>
+        Retrigger jobs
+      </Typography>
+      <Typography>
+        Choose how many clones of the base and new tasks will be started.
+      </Typography>
+      <form onSubmit={onFormSubmit}>
```

Comment:
Just curious. Why did you choose the native form vs the one supplied by react-router?

Reply:
Good question!

The one supplied by react-router is special: it comes with a lot of built-in functionality that interacts with react-router (for navigation especially, but also pushing data to a server).
For this case, we only use a `form` element to encapsulate the selects and the submit button, and use its `submit` event for accessibility (I believe screenreaders have a functionality to submit a form so that they don't have to `<tab>` to the button). We don't need and we don't want all the functionality that comes with react-router.

Hope this makes sense!